accepted/00000-replace-salt-toaster.md

```diff
@@ -0,0 +1,220 @@
+[design]: #detailed-design
+
+<!-- This is the bulk of the RFC. Explain the design in enough detail for
+somebody familiar with the product to understand, and for somebody familiar with
+the internals to implement. -->
+
+<!-- This section should cover architecture aspects and the rationale behind
+disruptive technical decisions (when applicable), as well as corner-cases and
+warnings. Whenever the new feature creates new user interactions, this section
+should include examples of how the feature will be used. -->
+
+## Overview
+
+The Salt upstream test suite is executed in virtual machines (VMs) that run the
+different operating systems we support. These VMs are set up with sumaform.
+Terracumber triggers terraform to create the VMs from within a Jenkins job and
+launches the test suite, connecting to each VM over SSH. Salt and the test suite
+are installed in the VM via a Salt Bundle that includes the tests.
+
+Pull Request tests are handled slightly different, using a Container containing
+the test dependencies. Management of the dependencies is not done through a
+Dockerfile that lists all dependencies, instead the dependencies are pulled in
+via RPM dependencies.
+
+## Provisioning Test VMs with Sumaform
+
+A new module "salt\_testenv" is added to sumaform. This module defines a new
+role, "salt\_testenv", and the default values needed to run all test suites,
+e.g. 4GiB of RAM. Sumaform installs the test suite from an RPM/Deb package.
+This package is, in most cases, a special Salt Bundle flavor that includes the
+test suite and its dependencies. Only for SLE 15, we additionally use a Salt
+sub-package that only contains the test suite. This sub-package pulls in the
+test dependencies via RPM dependencies.
+
+## Provisioning Test VMs for GitHub Actions
+
+GitHub Actions use so-called runners, which are VMs running Windows, macOS or
+Linux. It is possible to host your own runners, but we currently neither do not
+do that. We can't use sumaform to provision GitHub-hosted runners, so GH Actions
+will use a different approach.
+
+Since the GitHub-hosted Linux runners only run Ubuntu, we will use a Docker
+container that is based on openSUSE Leap (latest version). This container's
+Dockerfile installs a Salt sub-package that pulls in all required test
+dependencies. The container image can be build in OBS, which will automatically
+rebuilt it when included packages change.
+
+### Alteratives
+
+We could package the test suite and the dependencies as ".deb" packages instead
+of running a Leap-based container in GitHub Actions. This might be less overhead
+when running the tests, but adds the overhead of maintaining ".deb" packages of
+all test dependencies.
+
+Using the Salt Bundle flavor that we install in the Sumaform-provisioned test
+VMs could be an option, but then we would need to move the application and test
+code into the bundle.
+
+## Terracumber
+
+Terracumber is a tool used by our QE squad to operate terraform and cucumber for
```

Comment:
terracumber downloads sumform and run a terraform file with the definition we want, and passes a set of variables. IT can also be done without terracumber. 
It works as a help tool to reduce the load on configuring and managing the pipelines on jenkins.

Reply:
That's true. Do you think there is a better way to introduce terracumber?

Reply:
I think is fine as it is